infrastructure/helm-chart/charts/core/charts/upgrading/templates/scripts.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: upgrading-scripts
+data:
+  scale-down-deployment.sh: |
+    #!/bin/sh
+    deployment=${DEPLOYMENT_NAME}
+
+    kubectl scale deployment ${deployment} --replicas=0
+
+  scale-down-statefulset.sh: |
+    #!/bin/sh
+    statefulset=${STATEFULSET_NAME}
+
+    kubectl scale statefulset ${statefulset} --replicas=0
+
+  wait-for-empty-consumer-group.sh: |
+    #!/bin/sh
+    kafka_brokers=${KAFKA_BROKERS:-kafka:9092}
+    consumergroup=${CONSUMERGROUP:-1}
+    delay=${1:-10}
+    kafkas=$(echo ${kafka_brokers} | tr "," " ")
+
+    while ! kafka-consumer-groups.sh --bootstrap-server localhost:9092 --group api.CommunicationStores --describe 2>&1 | grep -q "has no active members"
+    do 
+      echo "Waiting for the consumer group to be empty..."
+      sleep ${delay}
+    done
+
+  reset-streaming-app.sh: |
+    #!/bin/sh
+    kafka_brokers=${KAFKA_BROKERS:-kafka:9092}
+    input_topics=${INPUT_TOPICS}
+    consumer_group=${CONSUMER_GROUP:-1}
+    delay=${1:-10}
+
+    kafka-streams-application-reset.sh --bootstrap-servers ${kafka_brokers} --to-earliest --application-id ${consumer_group} --input-topics ${input_topics}
```

Comment:
When we want to add a migration script, we need to create a file with the version (example 0.29.0.yaml) in the templates directory and add a block which creates the resources conditionally:
```
{{ if eq .Values.global.kubernetes.appImageTag "0.28.0" }}
  (bash scripts run from inside the cluster, using the upgrading-scripts as helper scrips)
{{ end }}
```

Reply:
That makes sense. How would one clean up the resources post migration?

Reply:
I guess we need to also do that inside the migration jobs (have a post-upgrade job that waits until stuff is done and cleans up). In case there needs to be a migration app that must run longer then the actual upgrade, then we can also deploy a `CronJob` that checks periodically and after the migration app needs to be deleted it also clears itself.

Reply:
Understood. For now, that seems like lots of added complexity only to, say, remove a Java pod. Perhaps it's ok to just include that part in the release notes for now. What do you think?

Reply:
I would leave the part for the migration/cleanup jobs until we actually have an upgrade with a migration. Then we will see with an actual example how this will work.
I think for all the migration resources we can add some specific annotations and at the end we can use a generic cleanup job.